File: eunji/0x05/BOJ_10828.cpp
#include <iostream>
#include <stack>
using namespace std;


int main(void){
    stack<int> S;
    int N;
    cin >> N;
    for (int i=0; i<N; i++){
        string order;
        cin >> order;
        if (order == "push"){
            int tar;
            cin >> tar;
            S.push(tar);
        }
        else if (order == "pop"){
            if (S.empty()) cout << -1 << '\n';
            else {
                cout << S.top() << '\n'; 
                S.pop();
                }
        }
        else if (order == "top"){
            if (S.empty()) cout << -1 << '\n';
            else cout << S.top() << '\n';
        }
        else if (order == "size"){
            cout << S.size() << '\n';
        }
        else if (order == "empty"){
            cout << S.empty() << '\n';          
        }
    }
}

File: eunji/0x05_stack/BOJ_10828.py
# from collections import deque
import sys

stack = []

N = int(sys.stdin.readline())
for i in range(N):
    orders = sys.stdin.readline().split()
    # print(orders)
    order = orders[0]
    if len(orders) == 2:
        tar = int(orders[1])
    if (order == "push"):
        stack.append(tar)
    elif (order == "pop"):
        if len(stack) > 0: print(stack.pop())
        else: print(-1)
    elif (order ==  "top"):
        if len(stack) > 0: print(stack[-1])
        else: print(-1)
    elif (order == "size"):
        print(len(stack))
    elif (order == "empty"):
        if len(stack) > 0: print(0)
        else: print(1)